api/src/index.js

```diff
@@ -29,3 +29,59 @@ app.use("/api", apiRouter);
 app.listen(process.env.PORT, () => {
   console.log(`API listening on port ${process.env.PORT}`);
 });
+//Future-meal
+app.get("/future-meals", async (req, res) => {
+  try {
+    const meals = await knex("Meal").where("when", ">", knex.fn.now());
+    res.json(meals);
+  } catch (error) {
+    res.status(500).json({ error: error.message });
+  }
+});
+
+//past-meal
+app.get("/past-meals", async (req, res) => {
+  try {
+    const now = new Date();
+    const meals = await knex("Meal").where("when", "<", now);
+    res.json(meals);
+  } catch (error) {
+    res.status(500).json({ error: error.message });
+  }
+});
+
+//all-meal
+app.get("/all-meals", async (req, res) => {
+  try {
+    const meals = await knex("Meal").orderBy("id");
+    res.json(meals);
```

Comment:
It is always a good idea to validate the result of knex query, else you risk sending SQL stacktrace to the frontend (not good)!